GliaWidgetsTests/SecureConversations/Confirmation/SecureConversations.ConfirmationViewModelTests.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import XCTest
+@testable import GliaWidgets
+
+final class SecureConversationsConfirmationViewModelTests: XCTestCase {
+    typealias ConfirmationViewModel = SecureConversations.ConfirmationViewModel
+    var viewModel: ConfirmationViewModel = .mock
+
+    override func setUp() {
+        viewModel = .mock
+    }
+}
+
+// Props
+extension SecureConversationsConfirmationViewModelTests {
+    func testPropsDoNotGenerateABackButton() {
+        let props = viewModel.props().confirmationViewProps.header
+
+        XCTAssertNil(props.backButton)
+    }
+
+    func testPropsGenerateCorrectTitle() {
+        let title = "Test"
+        var style = Theme().secureConversationsConfirmation
+        style.headerTitle = title
+
+        viewModel.environment = .init(confirmationStyle: style)
+
+        let props = viewModel.props()
+        XCTAssertEqual(props.confirmationViewProps.header.title, title)
+    }
+
+    func testPropsGenerateEndButtonWithAccessibilityIdentifier() {
+        let props = viewModel.props().confirmationViewProps.header.endButton
+
+        XCTAssertEqual(props.accessibilityIdentifier, "header_end_button")
+    }
+}
+
+// Delegate
+extension SecureConversationsConfirmationViewModelTests {
+    func testSendingCloseButtonEventCallsDelegate() {
+        viewModel.delegate = { event in
+            switch event {
+            case .closeTapped:
+                XCTAssertTrue(true)
+            default: XCTFail()
+            }
+        }
+
+        viewModel.event(.closeTapped)
```

Comment:
should be better to use `XCTestExpectation` here?
I am just worried, if sometime delegate will be async, this test will be succeeded because after line 51 everything in the test will be released, and since after this line no XCTFail Xcode will think this test is succeeded, or do I miss something?

Reply:
XCTestExpectation is actually not so good idea. All async processes in unit tests must to be taken under control.

Reply:
Yes for my welcome screen tests I was using expectation and then I was asked not to use it :) but how to figure out if the delegate stops working if it becomes async, I have no idea. Probably should be done by the person who makes that task at that point in time.

Reply:
@yurii-glia @gersonnoboa 
I would suggest to create placeholder variable, store result there, and then check against expected result:
```
var receivedEvent: SomeEventType? = ...
viewModel.delegate = { event in
   receivedEvent = event
}

switch XCTUnwrap(receivedEvent) {
      case .closeTapped:
             XCTAssertTrue(true)
      default: XCTFail()
 }

```

Note that `testSendingCloseButtonEventCallsDelegate ` has to be marked as _throwing_.

Reply:
agree with @igorkravchenko suggestion.